apps/api/schedule.py

```diff
@@ -134,6 +135,36 @@ def put(self, event_id):
         return {"is_favourite": new_state}
 
 
+class UpdateLotteryPreferences(Resource):
+    def post(self, proposal_type):
+        if proposal_type not in ["workshop", "youthworkshop"]:
+            abort(400)
+
+        if not current_user.is_authenticated:
+            abort(401)
+
+        new_order = [int(i) for i in request.get_json()]
+
+        current_tickets = {
+            t.id: t
+            for t in EventTicket.query.filter_by(
+                state="entered-lottery", user_id=current_user.id
+            ).all()
+            if t.proposal.type == proposal_type
+        }
+
+        for new_rank, t_id in enumerate(new_order):
+            ticket = current_tickets[t_id]
+            ticket.rank = new_rank
```

Comment:
Tickets which don't exist in the JSON but do exist in the DB will not be reranked, so you can get everything to rank 0 by submitting this multiple times with one ticket at a time.

I don't think this helps you at all since you'll still only win once, though.